File: bbb/bbb/report/printing_log_report/printing_log_report.py
# Copyright (c) 2013, Invento Software Limited and contributors
# For license information, please see license.txt

import frappe
from frappe import _
from erpnext.stock.report.stock_ledger.stock_ledger import get_item_group_condition


def execute(filters=None):
    columns = get_columns()
    data = get_items(filters)
    print(data)
    return columns, data


def get_columns():
    """return columns"""
    columns = [
        {"label": _("Date"), "fieldname": "date", "fieldtype": "date", "width": 140},
        {"label": _("Invoice Number"), "fieldname": "invoice_number", "fieldtype": "Link", "options": "POS Invoice",
         "width": 250},
        {"label": _("Location"), "fieldname": "location", "fieldtype": "Link", 'options': 'POS Profile', "width": 140},
        {"label": _("Served By"), "fieldname": "served_by", "fieldtype": "Link", 'options': 'Served By', "width": 140},
        {"label": _("Customer"), "fieldname": "customer", "fieldtype": "Link", "options":"Customer", "width": 170},
        # {"label": _("Customer Name"), "fieldname": "customer_name", "fieldtype": "Data", "width": 150},
        # {"label": _("Email"), "fieldname": "email", "fieldtype": "Data", "width": 150},
        # {"label": _("Mobile Number"), "fieldname": "mobile_number", "fieldtype": "Data", "width": 150},
        {"label": _("Total Print Count"), "fieldname": "total_print_count", "width": 140},
        {"label": _("Company"), "fieldname": "company", "width": 230},
    ]
    return columns


def get_conditions(filters):
    conditions = []

    if filters.get("from_date"):
        conditions.append("printing_log.date>='%s'" % filters.get("from_date"))
    if filters.get("to_date"):
        conditions.append("printing_log.date<='%s'" % filters.get("to_date"))
    if filters.get("served_by"):
        conditions.append("printing_log.served_by='%s'" % filters.get("served_by"))
    if filters.get("location"):
        conditions.append("printing_log.location='%s'" % filters.get("location"))
    if filters.get("customer"):
        conditions.append("printing_log.customer='%s'" % filters.get("customer"))
    if filters.get("company"):
        conditions.append("printing_log.company='%s'" % filters.get("company"))
    if conditions:
        conditions = " and ".join(conditions)
    return conditions


def get_items(filters):
    conditions = get_conditions(filters)

    query_result = frappe.db.sql(
        """select COUNT(ALL printing_log.invoice_number) as total_print_count, printing_log.date,
            printing_log.served_by, printing_log.location, printing_log.company,
            printing_log.invoice_number, printing_log.customer, printing_log.customer_name, printing_log.location
        from `tabPrinting Log` printing_log where {} group by printing_log.invoice_number""".format(conditions), as_dict=True)

    return query_result

File: bbb/bbb/report/printing_log_report/printing_log_report.js
// Copyright (c) 2022, invento software limited and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Printing Log Report"] = {
	"filters": [
		{
			"fieldname":"from_date",
			"label": __("From Date"),
			"fieldtype": "Date",
			"default": frappe.datetime.get_today(),
			"reqd": 1,
			"width": "60px"
		},
		{
			"fieldname":"to_date",
			"label": __("To Date"),
			"fieldtype": "Date",
			"default": frappe.datetime.get_today(),
			"reqd": 1,
			"width": "60px"
		},
		{
			"fieldname": "invoice_no",
			"label": __("Invoice Number"),
			"fieldtype": "Link",
			"width": "200",
			"options": "POS Invoice",
		},
		{
			"fieldname": "served_by",
			"label": __("Served By"),
			"fieldtype": "Link",
			"width": "80",
			"options": "Served By"
		},
		{
			"fieldname": "location",
			"label": __("Location"),
			"fieldtype": "Link",
			"width": "80",
			"options": "POS Profile"
		},
		{
			"fieldname": "customer",
			"label": __("Customer"),
			"fieldtype": "Link",
			"width": "80",
			"options": "Customer"
		},
		{
			"fieldname": "company",
			"label": __("Company"),
			"fieldtype": "Link",
			"width": "80",
			"options": "Company",
			"default": frappe.defaults.get_default("company")
		},
	]
};
